### ISSUE TITLE
How to use gspread when authenticating through google OAuth?

### ISSUE THREAD

**Issue by charanrajh** [burnash/gspread#664]
I'm developing a server application which fetches the rows from one private spread sheet and writes them to another private spreadsheet. Perhaps, I'm not using signed credentials method for authorization. I'm using google web server client id method in OAuth.
You can find it's [docs here.](https://developers.google.com/api-client-library/python/auth/web-app).
I used the same code in flask python framework. Now I have the credentials object. How to use this for gspread authorization. The object looks like the following:
![screenshot from 2019-02-26 17-21-53](https://user-images.githubusercontent.com/37415880/53410886-1beb1600-39eb-11e9-82e5-8f5adc93ecef.png)
When Iam passing this credentials object to gspread, it's throwing an error:
"__init__ has no access_token" [not the same vocabulary]

**Comment by nycnewman** [comment]
This is probably less than optimal but solved with the following code snippet. Issue appears to be that gspread does not support latest oAuth2 API (rather it still supports oauthapiclient). The newer API does not have the same schema as the old.

#Get a delegated credential using your service key
delegated_credentials = credentials.with_subject('<some user in your domain>')

from googleapiclient import _auth

# force system to get oAuth token (this seems to be done as part of discovery.build step in newer Google APIs
test = _auth.authorized_http(delegated_credentials)
# copy generated token into credentials
delegated_credentials.token = test.credentials.token
test2 = _auth.refresh_credentials(delegated_credentials)

# map token into old field name and copy other values across
class fixed_creds(service_account.Credentials):
    def __init__(self, creds):
        self.access_token = creds.token
        self.__dict__.update(creds.__dict__)

# create new credential object with the access_token
gcreds = fixed_creds(delegated_credentials)


As I said, this mayy be wrong way to achieve this but haven;t found a better way or had time to fully read up on new oAuth API.

**Comment by zmunro** [comment]
Anyone found a better way to do this? Or a better library that uses the up to date google auth?

**Comment by benjamindhimes** [comment]
The following works fine:
```python
import httplib2

import gspread
from oauth2client.client import flow_from_clientsecrets
from oauth2client.file import Storage
from oauth2client.tools import run_flow

CLIENT_SECRET = 'credentials.json'
SCOPE = 'https://www.googleapis.com/auth/spreadsheets.readonly'
STORAGE = Storage('credentials.storage')

# Start the OAuth flow to retrieve credentials
def authorize_credentials():
# Fetch credentials from storage
    credentials = STORAGE.get()
# If the credentials doesn't exist in the storage location then run the flow
    if credentials is None or credentials.invalid:
        flow = flow_from_clientsecrets(CLIENT_SECRET, scope=SCOPE)
        http = httplib2.Http()
        credentials = run_flow(flow, STORAGE, http=http)
    return credentials
credentials = authorize_credentials()

gs = gspread.authorize(credentials)
sheet = gs.open_by_url('https://docs.google.com/spreadsheets/d/1BGG6isA4wt1pN0P_WV8I-dURTTj86_PdxaaUzgmL-6w/edit?usp=sharing')
wks = sheet.sheet1
print(wks.get_all_records())
```

**Comment by taewookim** [comment]
Having same issue. How do I use the refresh token from Google auth and pass it to gspread?

**Comment by taewookim** [comment]
I found the solution: https://github.com/burnash/gspread/issues/472#issuecomment-317631725

**Comment by Joohansson** [comment]
Yeah the 472 comment above worked for me. Modified to work with scope and private key.

```
import gspread
from google.auth.transport.requests import AuthorizedSession
from google.oauth2 import service_account

googleAPI = '/pathToSecret.json'
scope = ['https://www.googleapis.com/auth/drive']
credentials = service_account.Credentials.from_service_account_file(googleAPI)
scopedCreds = credentials.with_scopes(scope)
gc = gspread.Client(auth=scopedCreds)
gc.session = AuthorizedSession(scopedCreds)
sheet = gc.open("MyDoc").sheet1
```

**Comment by Lucas1Jorge** [comment]
> Yeah the 472 comment above worked for me. Modified to work with scope and private key.
> 
> ```
> import gspread
> from google.auth.transport.requests import AuthorizedSession
> from google.oauth2 import service_account
> 
> googleAPI = '/pathToSecret.json'
> scope = ['https://www.googleapis.com/auth/drive']
> credentials = service_account.Credentials.from_service_account_file(googleAPI)
> scopedCreds = credentials.with_scopes(scope)
> gc = gspread.Client(auth=scopedCreds)
> gc.session = AuthorizedSession(scopedCreds)
> sheet = gc.open("MyDoc").sheet1
> ```

Thank you very much. It works!

**Comment by KarthikRajashekaran** [comment]
once the Secret.json is created, How can I use it via AWS SecretManager instead of the JSON File
